scipy/stats/_axis_nan_policy.py

```diff
@@ -183,30 +183,41 @@ def _masked_arrays_2_sentinel_arrays(samples):
 
     # Choose a sentinel value. We can't use `np.nan`, because sentinel (masked)
     # values are always omitted, but there are different nan policies.
+    dtype = np.result_type(*samples)
+    dtype = dtype if np.issubdtype(dtype, np.number) else np.float64
```

Comment:
`else np.float64` was intended to be for types that aren't NumPy numbers but can be converted.